compiler/noirc_evaluator/src/ssa/opt/unrolling.rs

```diff
@@ -302,6 +302,11 @@ impl Loop {
     pub(super) fn get_const_upper_bound(&self, function: &Function) -> Option<FieldElement> {
         let block = &function.dfg[self.header];
         let instructions = block.instructions();
+        if instructions.is_empty() {
+            // If the loop condition is constant time, the loop header will be
+            // simplified to a simple jump.
```

Comment:
So this loop will have zero iterations, right, there is nothing whatsoever to unroll?

Reply:
Just curious:would it work if we returned `Some(0)` instead of `None`?

Reply:
That'd set the upper bound at zero which could work differently if we ever allow looping starting from a negative index: `for _ in -5 .. 0 { ... }`. Returning None is more correct here